### DISCUSSION TITLE
what's the behavior when getServerSideProps and getInitialProps both exist?

### DISCUSSION THREAD

**Discussion by yxwu** [vercel/next.js#13297] [Help]
As stated in title, in this case, is this SSR or SSG? Will their output be merged?

And how about getStaticProps and getInitialProps both exist?

Thanks

**Comment by Timer** [comment]
They are not allowed to co-exist and Next.js will throw an error saying you must only use one.

**Reply by yxwu** [reply]
Thanks for reply.
Yes, they can not coexist in the same page file, such as index.js.
But what if getInitialProps exists in _app.js, and at the same time getStaticProps exist in index.js? Seems in this case, SSG still works (static html + json), I assume this is a valid use case, am I correct?
By the way, seems we can NOT put getStaticProps / getServerSideProps inside _app.js, right?
